gems/smithy/lib/smithy/weld.rb

```diff
@@ -33,16 +33,26 @@ def post_process(artifacts)
       artifacts
     end
 
-    # Called when constructing a plan
+    # Called when constructing endpoint parameters. Any bindings defined here will
+    # be merged with other built-in bindings.
     # @return [Hash<String, Hash>] endpoint built in bindings for use in endpoint rules.
     def endpoint_built_in_bindings
       {}
     end
 
-    # Called when constructing a plan
+    # Called when constructing the endpoint provider. Any bindings defined here will
+    # be merged with other function bindings.
     # @return [Hash<String, String>] endpoint function bindings for use in endpoint rules.
     def endpoint_function_bindings
       {}
     end
+
+    # Called when constructing the client. Any plugins defined here will be merged
```

Comment:
we'll also want the ability to add dependencies to the gemspec, but no need to put it in now.

Reply:
Yes, we need a way to add to the gemspec. We could even create a gemspec object and pass it to weld, then add to it directly, then write it out.

Reply:
oh yeah, a gemspec object would be awesome for that! but we can figure that out when we get there w/ AWS I think.